frontend/app/src/pages/price-manager/historic/index.vue

```diff
@@ -26,11 +32,60 @@ const valid = ref(false);
 const editMode = ref(false);
 
 const { setMessage } = useMessageStore();
-const { editHistoricalPrice, addHistoricalPrice } = useAssetPricesApi();
+const { editHistoricalPrice, addHistoricalPrice, fetchHistoricalPrices } =
+  useAssetPricesApi();
 const router = useRouter();
 const route = useRoute();
 const { tc } = useI18n();
 
+const prices: Ref<HistoricalPrice[]> = ref([]);
+const loading: Ref<boolean> = ref(false);
+const { notify } = useNotificationsStore();
+
+const fetchPrices = async (payload?: Partial<ManualPricePayload>) => {
+  set(loading, true);
+  try {
+    set(prices, await fetchHistoricalPrices(payload));
+  } catch (e: any) {
+    const notification: NotificationPayload = {
+      title: tc('price_table.fetch.failure.title'),
+      message: tc('price_table.fetch.failure.message', 0, {
+        message: e.message
+      }),
+      display: true,
+      severity: Severity.ERROR,
+      category: NotificationCategory.DEFAULT
+    };
+    notify(notification);
+  } finally {
+    set(loading, false);
+  }
+};
+
+const { resetHistoricalPricesData } = useHistoricCachePriceStore();
+
+const refresh = async (modified: boolean | HistoricalPrice = false) => {
```

Comment:
Why do that instead of making modified optional?

Reply:
so there are 3 state of refresh
1. modified = false, refresh without modify anything from `useHistoricCachePriceStore`
2. modified = true, remove cache from all data we have on historic price table
3. modified = Historical Price, do no 2 + delete cache for specific key (happens when we delete historic price, where the item is not exist anymore on the table, but still need to be cleared)

for no 1, yes we can make it optional

Reply:
like this
```
const refresh = async (modified?: true | HistoricalPrice) => {
```

Reply:
> Can you use two parameters here instead of this unholy union 😄

Just saw the edited version, yeah I will make it 2 parameters

Reply:
Yes I was reading the code and saw that it was 3 states so I updated it :)